src/player.rs

```diff
@@ -430,7 +450,59 @@ impl PlayerInternal {
         let key = self.session.audio_key().request(track.id, file_id).wait().unwrap();
 
         let encrypted_file = AudioFile::open(&self.session, file_id).wait().unwrap();
-        let audio_file = Subfile::new(AudioDecrypt::new(key, encrypted_file), 0xa7);
+
+        let mut decrypted_file = AudioDecrypt::new(key, encrypted_file);
+
+        let mut normalization_factor: f32 = 1.0;
+
+        if self.config.normalization {
+            //buffer for float bytes
```

Comment:
Break this up into a function which does the parsing and returns a struct with 4 f32s. You can `#[derive(Debug)]` and use a single debug statement.